### PR TITLE
Add checksum (md5) function.

### PR DESCRIPTION
**Pull request by mponomar**
Found this useful for my own db - function checksums the input string.  If input is not a string, returns NULL.

### CODE REVIEW

File: sqlite/src/func.c [hunk 1]
```diff
@@ -24,6 +24,8 @@
 #include "comdb2.h"
 #include "sql.h"
 #include "bdb_int.h"
+#include "md5.h"
```
Comment (nirbhayc) [inline sqlite/src/func.c]: Enclose the changes in WITH_SSL ?

Comment (mponomar) [reply sqlite/src/func.c]: This is SQLite's MD5 implementation, not openssl's.  We use it in other places (eg: fingerprint code), even without WITH_SSL.

Comment (nirbhayc) [reply sqlite/src/func.c]: oh ok. I looked at the signature thought they were openssl's.

File: sqlite/src/func.c [hunk 2]
```diff
@@ -971,6 +973,26 @@ static void comdb2StartTimeFunc(
...
+  sqlite3_context *context,
+  int argc,
+  sqlite3_value **argv
+  ){
+    const unsigned char *s;
+    MD5Context ctx = {0};
+    char checksum[16];
+    char out[33];
+
+    if (sqlite3_value_type(argv[0]) != SQLITE_TEXT)
```
Comment (mistachkin) [inline sqlite/src/func.c]: It would also be nice to be able to perform the md5sum() function on a blob value.

**Pull request merged**